api/v1beta1/detection/config.go

```diff
@@ -0,0 +1,168 @@
+package detection
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// DetectorConfig provides type-safe access to detector-specific configuration.
+// Detectors should use default values for missing configuration.
+type DetectorConfig interface {
+	// GetInt retrieves an integer configuration value
+	GetInt(key string, defaultValue int) int
+
+	// GetString retrieves a string configuration value
+	GetString(key string, defaultValue string) string
+
+	// GetBool retrieves a boolean configuration value
+	GetBool(key string, defaultValue bool) bool
+
+	// GetFloat64 retrieves a float64 configuration value
+	GetFloat64(key string, defaultValue float64) float64
+
+	// Has checks if a configuration key exists
+	Has(key string) bool
+}
+
+// detectorConfig implements DetectorConfig with smart type conversion
+type detectorConfig struct {
+	data map[string]any
+}
+
+// NewDetectorConfig creates a new DetectorConfig from a map
+func NewDetectorConfig(data map[string]any) DetectorConfig {
```

Comment:
`NewDectorConfig(nil)` looks weird, we should enforce to always have a map here, specially because we have a `NewEmptyDetectorConfig()` for cases where no map is passed.